Homework-12/script.js

```diff
@@ -0,0 +1,193 @@
+//   if (err) {
+//     console.log(err);
+//   } else
+//     loadScript("test2.js", (err, script) => {
+//       if (err) {
+//         console.log(err);
+//       } else
+//         loadScript("test3.js", (err, script) => {
+//           if (err) {
+//             console.log(err);
+//           } else
+//             loadScript("test4.js", (err, script) => {
+//               if (err) {
+//                 console.log(err);
+//               } else
+//                 loadScript("test5.js", (err, script) => {
+//                   if (err) {
+//                     console.log(err);
+//                   } else console.log("calback finished");
+//                 });
+//             });
+//         });
+//     });
+// });
+
+// Task 2
+
+// Можно ли "перевыполнить" промис?
+
+// let promise = new Promise(function (resolve, reject) {
+//   resolve(1);
+
+//   setTimeout(() => resolve(2), 1000); // never succeed
+// });
+
+// promise.then(alert); // 1
+
+// Task 3
+
+// Задержка на промисах
+// Встроенная функция setTimeout использует колбэк-функции. Создайте альтернативу, использующую промисы.
+
+// Функция delay(ms) должна возвращать промис, который перейдёт в состояние «выполнен» через ms миллисекунд, так чтобы мы могли добавить к нему .then:
+
+// function delay(ms) {
+//   return new Promise((resolve, reject) => {
+//     setTimeout(() => resolve(1), ms);
+//   })
+// }
+
+// delay(3000).then(() => console.log('выполнилось через 3 секунды'));
+
+// Task 4
+
+// Промисы: сравните then и catch
+// Являются ли фрагменты кода ниже эквивалентными? Другими словами, ведут ли они себя одинаково во всех обстоятельствах, для всех переданных им обработчиков?
+
+// promise.then(f1).catch(f2); // if promise.state === fulfield => then(f1). promise.state === rejected => cath(f2)
+
+// // Против:
+
+// promise.then(f1, f2); // if promise.state === fulfield => then(f1, f2). promise.state === rejected => unhandledRejection
+
+// Task 5
+
+// Ошибка в setTimeout
+// Что вы думаете? Выполнится ли .catch? Поясните свой ответ.
+
+// new Promise(function(resolve, reject) {
+//   setTimeout(() => {
+//     throw new Error("Whoops!");
+//   }, 1000);
+// }).catch(console.log("err")); // => err => Error: Whoops!
+
+// Task 6
+
+// Перепишите, используя async/await
+// Перепишите один из примеров раздела Цепочка промисов, используя async/await вместо .then/catch:
+
+// function loadJson1(url) {
+//   return fetch(url).then((response) => {
+//     if (response.status == 200) {
+//       return response.json();
+//     } else {
+//       throw new Error(response.status);
+//     }
+//   });
+// }
+
+// loadJson1("no-such-user.json") // (3)
+//   .catch(alert); // Error: 404
+
+// async function loadJson2(url) {
+//   let result = await fetch(url);
+//   return result((response) => {
+//     if (response.status == 200) {
+//       return response.json();
+//     } else {
+//       throw new Error(response.status);
+//     }
+//   });
+// }
```

Comment:
![image](https://user-images.githubusercontent.com/25768503/79700002-87511080-829b-11ea-8cf0-eb329e220e7b.png)

1. let -> const
2. space between logical entities
3. `return response.json();`  -> resolve it and return result.